### ISSUE TITLE
Inference on set of images

### ISSUE THREAD

**Issue by gaussiangit** [wvangansbeke/Unsupervised-Classification#35]
How can we use this to cluster similar images in a given folder with pretrained imagenet model ?

**Comment by wvangansbeke** [comment]
Hi @gaussiangit,

You will have to to load your own images from your directory. Then you can load our pretrained ImageNet weights and pass the images trough the model to get the predictions.  I would simply write a for loop to load the images 1 by 1 but if you have a large dataset, you might want to use the dataloader from Pytorch to make it multi-threaded.

**Comment by gaussiangit** [comment]
Thank you for quick response 👍 @wvangansbeke

**Comment by gaussiangit** [comment]
@wvangansbeke Some basic question. Do I have to apply the same transformations that you used in the repo on the test set images while evaluating ? I am just doing the predictions on pretrained model as you have mentioned earlier.

**Comment by wvangansbeke** [comment]
Hi @gaussiangit,

Yes. You can probably use the same augmentations as we use during validation (see [this line](https://github.com/wvangansbeke/Unsupervised-Classification/blob/master/utils/common_config.py#L247). The augmentations `ToTensor` and `Normalize` are mandatory actually.

**Comment by dabasmoti** [comment]
> Hi @gaussiangit,
> 
> You will have to to load your own images from your directory. Then you can load our pretrained ImageNet weights and pass the images trough the model to get the predictions. I would simply write a for loop to load the images 1 by 1 but if you have a large dataset, you might want to use the dataloader from Pytorch to make it multi-threaded.

Can you please give an example on how to do it?
